src/utilities/retry.ts

```diff
@@ -0,0 +1,58 @@
+import { timeout } from "./timeout"
+
+type RetryCallback<T> = () => T
+
+export type RetryOptions = {
+  count: number,
+  delay: number,
+}
+
+export const DEFAULT_RETRY_OPTIONS: RetryOptions = {
+  count: 0,
+  delay: 500,
+}
+
+function normalizeRetryOptions(options: Partial<RetryOptions> | number | undefined): RetryOptions {
+  if(typeof options === 'number') {
+    return {
+      ...DEFAULT_RETRY_OPTIONS,
+      count: options,
+    }
```

Comment:
_[nit]_
similar suggest that I made on the last one, bit time nit, fine if you don't like it

```suggestion
     return normalizeRetryOptions({count: options})
```